beacon-chain/core/helpers/attestation.go

```diff
@@ -191,7 +192,13 @@ func ValidateAttestationTime(attSlot primitives.Slot, genesisTime time.Time, clo
 	}
 	attSlotEpoch := slots.ToEpoch(attSlot)
 	if attSlotEpoch != currentEpoch && attSlotEpoch != prevEpoch {
-		return fmt.Errorf("attestation slot %d not within current epoch %d or previous epoch %d", attSlot, currentEpoch, prevEpoch)
+		attError = fmt.Errorf(
+			"attestation epoch %d not within current epoch %d or previous epoch %d",
+			attSlot/params.BeaconConfig().SlotsPerEpoch,
+			currentEpoch,
+			prevEpoch,
+		)
+		return errors.Join(ErrIncorrectEpoch, attError)
```

Comment:
We should add the same `ErrTooLate` here, as we check for this in gossip. If a message has this particular error, we skip logging it.

Reply:
[beacf99](https://github.com/prysmaticlabs/prysm/pull/12813/commits/beacf990e94af815b6e41b9f1358a7b2c9c8df0f)